prefect_utils/email_unsubscribes.py

```diff
@@ -0,0 +1,74 @@
+
+import backoff
+import requests
+from prefect import task
+from sailthru.sailthru_client import SailthruClient
+
+from . import snowflake
+
+
+@task
+@backoff.on_exception(backoff.expo,
+                      snowflake.connector.ProgrammingError,
+                      max_tries=3)
+def sync_sailthru_to_braze(
+    sf_credentials: dict,
+    sf_role: str,
+    sf_database: str,
+    sf_schema: str,
+    sf_table: str,
+    braze_api_key: str,
+    braze_api_server: str,
+):
+    sf_connection = snowflake.create_snowflake_connection(
+        sf_credentials,
+        sf_role,
+    )
+    query = """
+    SELECT email FROM {table}
+    """.format(
+        table=snowflake.qualified_table_name(sf_database, sf_schema, sf_table),
+    )
+    cursor = sf_connection.cursor()
+    cursor.execute(query)
+    # 50 email batches because that's how many braze accepts at a time
+    for rows in cursor.fetchmany(50):
+        requests.post(
+            url=f"{braze_api_server}/email/status",
+            json={
+                'email': [row['email'] for row in rows],
+                'subscription_state': 'unsubscribed',
+            },
+            headers={'Authorization': f'BEARER {braze_api_key}'}
+        )
+
+
+@task
+@backoff.on_exception(backoff.expo,
+                      snowflake.connector.ProgrammingError,
```

Comment:
Ditto here - except for the Sailthru client instead of Braze API?

Reply:
Fixed.